homeassistant/core.py

```diff
@@ -218,7 +217,12 @@ def async_add_job(self, target: Callable[..., None], *args: Any) -> None:
 
         # if a task is sheduled
         if task is not None:
-            self._pending_tasks.add(task)
+            self._pending_tasks.append(task)
+
+        # cleanup
+        if len(self._pending_tasks) > 50:
```

Comment:
How did you pick 50 ? We already had people that had trouble on Windows because they went > 512 tasks. I wonder if it would be better to just schedule a cleanup every 10 seconds ?